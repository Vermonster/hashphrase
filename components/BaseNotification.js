import React from 'react';
import {
  Card, CardItem, Body, Text,
} from 'native-base';

const BaseNotification = ({ message }) => (
  <Card>
    <CardItem>
      <Body>
        <Text>
          { message }
        </Text>
      </Body>
    </CardItem>
  </Card>
);

export default BaseNotification;
