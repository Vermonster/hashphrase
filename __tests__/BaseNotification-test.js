import React from 'react';
import { shallow } from 'enzyme';
import { Text, Card } from 'react-native-paper';
import BaseNotification from '../src/components/BaseNotification';

const innerText = 'Testing message';

describe('<BaseNotification />', () => {
  it('should match the existing snapshot', () => {
    const wrapper = shallow(<BaseNotification message={innerText} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the the correct components', () => {
    const requiredComponents = [Text, Card];
    const wrapper = shallow(<BaseNotification message={innerText} />);
    requiredComponents.forEach((component) => {
      expect(wrapper.find(component)).toHaveLength(1);
    });
  });

  it('should render message prop within nested Text component', () => {
    const notification = 'This is a test notification';
    const wrapper = shallow(<BaseNotification message={notification} />);
    const text = wrapper.find(Text);
    expect(text.props().children).toEqual(notification);
  });
});
