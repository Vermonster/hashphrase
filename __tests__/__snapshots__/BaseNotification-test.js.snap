// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BaseNotification /> should match the existing snapshot 1`] = `
<Styled(Card)>
  <Styled(CardItem)>
    <Styled(Body)>
      <Styled(Text)>
        Testing message
      </Styled(Text)>
    </Styled(Body)>
  </Styled(CardItem)>
</Styled(Card)>
`;
